backend/dao/usersDAO.js

```diff
@@ -0,0 +1,79 @@
+const ObjectID = require("mongoose").ObjectID;
+
+//This is a variable to use to store a reference to the db
+let users;
+
+//* injectDB method. (to initially conenct to db)
+//* as soon as server starts, get a reference to the users database.
+//* if it is already filled, then return, but if not already filled, then assign the variable with a reference to the db.
+module.exports = class UsersDAO {
+  static async injectDB(conn) {
+    if (users) {
+      return;
+    }
+    try {
+      //*connect to the db 'sample_analyrics' then access the collection 'customers'.
+      users = await conn.db(process.env.USER_NS).collection("customers");
+    } catch (e) {
+      console.error(`Unable to establish connection in usersDAO: ${e}`);
+    }
+  }
+  //Then, run the function get
+  //?call this to get a list of all users in the db
+  //?call this method to query/sort
+  //* corresponds to the .get
+  static async getUsers({
+    //can put many queries here..
+    //https://docs.mongodb.com/manual/reference/operator/
+    filters = null,
+    page = 0,
+    usersPerPage = 20,
+  } = {}) {
+    let query;
+    if (filters) {
+      if ("name" in filters) {
+        //this filter needed to be set up in Atlas, in 'create index'
+        query = { $text: { $search: filters["name"] } };
+      } else if ("email" in filters) {
+        query = { email: { $eq: filters["email"] } };
+      } else if ("accounts" in filters) {
+        query = { accounts: { $eq: filters["accounts"] } };
+      }
+    }
+    let cursor;
+    try {
+      cursor = await users.find(query); //*find all users from db that goes along with query that was passed in.
+    } catch (e) {
+      console.error(`Unable to issue find query: ${e}`);
+      return { usersList: [], totalNumUsers: 0 };
+    }
+    const displayCursor = cursor
+      .limit(usersPerPage) //* user per page

```

Comment:
One thing I noticed:
You are over commenting 😅 
If a comment describes exactly what the code does, like: 
`find all users from db that goes along with query that was passed in.` for `await users.find(query)` or
`user per page` for `.limit(usersPerPage)`, we can totally skip the comment, the code itself expresses the same thing the comment says.
Rule of thumb about comments:
Comments should exist only to explain `why` a certain snippet of code exists, the `how` and `what` (how and what data is computed) is expressed by the code itself.
If it's not clear what a snippet of code is doing, then it should be re-written in a simpler way so that anyone can understand what is happening.
Many times the `why` is not clear just looking at the code, example is handling weird edge cases or working around bugs from third party libraries, in these cases it's valuable to write comments to let everyone know why certain code exists.
For example we don't need to explain the reason why we pass the query object with the filters, it's a pretty common pattern when implementing a controller, but the special `text` filter is a good comment.
Also we can have an exception for all I just said when the logic is intrinsically so complex that is very difficult to explain the `how` and `what` with just code, then we can add comments to guide anyone reading it.

Reply:
thanks for the tips on commenting! At the time, I was writing notes for myself, since I am still learning, and I was thinking Timea could use the notes if she was looking through and trying to understand it as well.. I will definitley clean up the comments once I finish setting it up